### PR TITLE
adding autoprefixer to build process

### LINKED ISSUE #24: Add autoprefixer

### PR DESCRIPTION
**Pull request by gvn**
closes #24

### CODE REVIEW

**Comment by Pomax** [comment]
I remember autprefixer being an absolute jerk on windows before (using bash-specific syntax that caused files like `$1` to be written to disk instead of acting as shell variable, etc). Hopefully they fixed that.

**Comment by Pomax** [comment]
yeah, that doesn't seem to make windows very happy.
```
> mofo-ui@1.5.1 build:scss c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui
> node-sass src/main.scss dist/mofo-ui.css && node-sass src/demo/demo.scss demo/css/demo.css

Rendering Complete, saving .css file...
Wrote CSS to c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui\dist\mofo-ui.css
Rendering Complete, saving .css file...
Wrote CSS to c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui\demo\css\demo.css

> mofo-ui@1.5.1 build:autoprefix c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui
> postcss --use autoprefixer --autoprefixer.browsers 'last 3 versions' -o dist/mofo-ui.css dist/mofo-ui.css

Usage: C:\Program Files\nodejs\node.exe
c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui\node_modules\postcss-cli-simple\bin\postcss -use plugin
[--config|-c config.json] [--output|-o output.css] [input.css]

Options:
  -c, --config       JSON file with plugin configuration
  -u, --use          postcss plugin name (can be used multiple times)
  -o, --output       Output file
  -m, --map          Source map
  -s, --syntax       Alternative input syntax parser
  -p, --parser       Alternative CSS parser
  -t, --stringifier  Alternative output stringifier
  -v, --version      Show version number                                                                  [boolean]
  -h, --help         Show help                                                                            [boolean]

Examples:
  postcss --use autoprefixer -c options.json -o screen.css   Use autoprefixer as a postcss plugin
  screen.css
  postcss --use autoprefixer --autoprefixer.browsers "> 5%"  Pass plugin parameters in plugin.option notation
  -o screen.css screen.css

Please specify a single input file.

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build:autoprefix"
npm ERR! node v6.2.2
npm ERR! npm  v3.9.5
npm ERR! code ELIFECYCLE
npm ERR! mofo-ui@1.5.1 build:autoprefix: `postcss --use autoprefixer --autoprefixer.browsers 'last 3 versions' -o dist/mofo-ui.css dist/mofo-ui.css`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mofo-ui@1.5.1 build:autoprefix script 'postcss --use autoprefixer --autoprefixer.browsers 'last 3 versions' -o dist/mofo-ui.css dist/mofo-ui.css'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mofo-ui package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     postcss --use autoprefixer --autoprefixer.browsers 'last 3 versions' -o dist/mofo-ui.css dist/mofo-ui.css
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mofo-ui
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mofo-ui
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui\npm-debug.log

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
npm ERR! node v6.2.2
npm ERR! npm  v3.9.5
npm ERR! code ELIFECYCLE
npm ERR! mofo-ui@1.5.1 build: `npm run build:js && npm run build:scss && npm run build:autoprefix && npm run build:demo`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mofo-ui@1.5.1 build script 'npm run build:js && npm run build:scss && npm run build:autoprefix && npm run build:demo'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mofo-ui package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run build:js && npm run build:scss && npm run build:autoprefix && npm run build:demo
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mofo-ui
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mofo-ui
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui\npm-debug.log

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! node v6.2.2
npm ERR! npm  v3.9.5
npm ERR! code ELIFECYCLE
npm ERR! mofo-ui@1.5.1 start: `npm i && npm run build && npm-run-all --parallel server watch`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mofo-ui@1.5.1 start script 'npm i && npm run build && npm-run-all --parallel server watch'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mofo-ui package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm i && npm run build && npm-run-all --parallel server watch
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mofo-ui
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mofo-ui
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     c:\Users\Mike\Documents\git\collaborations\mozilla\mofo-ui\npm-debug.log
```

File: package.json [hunk 1]
```diff
@@ -8,10 +8,11 @@
     "npm": "^3.0.0"
   },
   "scripts": {
+    "build:autoprefix": "postcss --use autoprefixer --autoprefixer.browsers 'last 3 versions' -o dist/mofo-ui.css dist/mofo-ui.css",
```
Comment (Pomax) [inline package.json]: due to how not-cross-platform autoprefixer is, this only works *without* the `--autoprefixer.browsers 'last 3 versions'` part: the browser version flag relies on bash-specific syntax that breaks all the things in both windows command shell as well as Powershell =(

**Comment by gvn** [comment]
@Pomax I refactored it as a Node script...give this a shot!

**Comment by Pomax** [comment]
nice! no errors when running this. R+

**Pull request merged**